docs/modules/ROOT/pages/join-two-streams.adoc

```diff
@@ -0,0 +1,391 @@
+
+The sample streaming data is adapted from this link:https://developer.confluent.io/tutorials/join-a-stream-to-a-stream/ksql.html[tutorial for Confluent Platform].
+
+== Before you Begin
+
+Before starting this tutorial, make sure that you have the following prerequisites:
+
+* link:https://docs.docker.com/compose/install/[Docker Compose]
+* link:https://git-scm.com/book/en/v2/Getting-Started-Installing-Git[Git]
+
+== Step 1. Clone the Project
+
+To set up the project, you need to download the code from GitHub.
+
+. Clone the GitHub repository.
++
+[tabs] 
+====
+HTTPS:: 
++ 
+--
+```bash
+git clone https://github.com/hazelcast-guides/stream-to-stream-joins.git
+cd stream-to-stream-joins
+```
+--
+SSH:: 
++ 
+--
+```bash
+git clone git@github.com:hazelcast-guides/stream-to-stream-joins.git
+cd stream-to-stream-joins
+```
+--
+====
+
+== Step 2. Start the Docker Containers
+
+In this step, you'll use Docker Compose to start all the Docker containers, including a Kafka broker, Hazelcast Platform, and Management Center.
+
+```bash
+docker compose up -d
+```
+
+You should see the following:
+
+```
+[+] Running 7/7
+ ⠿ Container zookeeper          Started                                                                                                                      0.7s
+ ⠿ Container management-center  Started                                                                                                                      0.6s
+ ⠿ Container hazelcast          Started                                                                                                                      0.7s
+ ⠿ Container broker             Started                                                                                                                      1.2s
+ ⠿ Container schema-registry    Started                                                                                                                      1.8s
+ ⠿ Container ksqldb-server      Started                                                                                                                      2.6s
+ ⠿ Container ksqldb-cli         Started                                                                                                                      3.3s
+```
+
+The Docker containers are running in detatched mode. You can see that they are running, using the following command:
+
+```
+docker ps
+```
+
+To see the logs of your Hazelcast member, use the following command:
+
+```
+docker logs hazelcast
+```
+
+You should see that you have a single member running in the cluster.
+
+```
+Members {size:1, ver:1} [
+	Member [172.19.0.4]:5701 - 15116025-342b-43c0-83f7-a2a90f0281ce this
+]
+```
+
+== Step 3. Create Two Kafka Streams
+
+To create the Kafka stream, you'll use KSQL, which is a streaming SQL engine for Apache Kafka. To submit the KSQL commands, you'll use the ksqlDB CLI.
```

Comment:
We see KSQL as a direct competitor to our SQL engine. If I understand correctly we use it here to insert some JSON data into a kafka topic.

We could achieve the same using just plain kafka and kafka-console-producer. See e.g here https://stackoverflow.com/a/54673559/951609

If we want to avoid downloading kafka distribution we can use the `cp-kafka` image which contains the script.

```
$ docker run confluentinc/cp-kafka:7.1.0 kafka-console-producer
Missing required option(s) [bootstrap-server]
Option                                   Description                            
------                                   -----------                            
--batch-size <Integer: size>             Number of messages to send in a single 
                                           batch if they are not being sent     
                                           synchronously. (default: 200)        
--bootstrap-server <String: server to    REQUIRED unless --broker-list          
  connect to>                              (deprecated) is specified. The server
                                           (s) to connect to. The broker list   
                                           string in the form HOST1:PORT1,HOST2:
                                           PORT2.                        
```

Reply:
As a side effect, you would avoid downloading a lot of data for the docker-compose.